src/python/pants/testutil/pants_integration_test.py

```diff
@@ -117,6 +117,13 @@ def run_pants_with_workdir_without_waiting(
             fp.write(TomlSerializer(config).serialize())
         args.append(f"--pants-config-files={toml_file_name}")
 
+    # The python backend requires setting ICs explicitly.
+    # We do this centrally here for convenience.
+    if any("pants.backend.python" in arg for arg in command) and not any(
+        "--python-interpreter-constraints" in arg for arg in command
+    ):
+        args.append("--python-interpreter-constraints=['>=3.7,<4']")
```

Comment:
Ditto a deprecation here? Or are there too many callsites to actually fix?

Reply:
This is too hacky not to fix, but it will take time. I'll try and see how many tests are affected.